### PR TITLE
refactor(gclient): avoid async calls inside into_iter() loop

### PR DESCRIPTION
**Pull request by ukint-vs**
Since awaiting inside into_iter() loop can't be `Send` we can fetch mailbox messages before and filter.

@gear-tech/dev

### CODE REVIEW

File: gclient/src/api/calls.rs [hunk 1]
```diff
@@ -753,15 +754,17 @@ impl GearApi {
...
+            args.clone().into_iter().map(|(mid, _, _, _)| mid).collect();
+        let mailbox_messages = self
+            .get_mailbox_messages(100)
+            .await?
+            .into_iter()
+            .map(|(message, _interval)| (message.id(), message.value()));
+        let mut values: BTreeMap<MessageId, Option<u128>> = mailbox_messages
+            .filter(|(mid, _value)| message_ids.contains(mid))
+            .map(|(mid, value)| (mid, Some(value)))
+            .collect();
```
Comment (breathx) [inline gclient/src/api/calls.rs:767]:
```suggestion
        let message_ids: Vec<_> =
            args.clone().into_iter().map(|(mid, _, _, _)| mid).collect();
        let mailbox_messages = self
            .get_mailbox_messages(message_ids.len()) // maybe? or x5 something like that
            .await?
            .into_iter()
            .map(|(message, _interval)| (message.id(), message.value()));
        let mut values: BTreeMap<_, _> = mailbox_messages
            .filter(|(mid, _value)| message_ids.contains(mid))
            .collect();
```

**Review by breathx — CHANGES_REQUESTED** [review]
This take of "all" messages from mailbox isn't optimal and may cause errors often. I'd go with query one-by-one anyway

File: gclient/src/api/calls.rs [hunk 2]
```diff
@@ -533,15 +533,18 @@ impl GearApi {
...
-        let mut values = BTreeMap::new();
-        for message_id in args.clone().into_iter() {
-            values.insert(
-                message_id,
-                self.get_mailbox_message(message_id)
-                    .await?
-                    .map(|(message, _interval)| message.value()),
-            );
-        }
+        let message_ids: Vec<MessageId> = args.clone().into_iter().collect();
```
Comment (breathx) [inline gclient/src/api/calls.rs:536]: for all changes auto-guessing types still valid 
```suggestion
        let message_ids: Vec<_> = args.clone().into_iter().collect();
//      or
//      let message_ids = args.clone().into_iter().collect::<Vec<_>>();
```

**Review by breathx — APPROVED** [review]
Awesome!

**Review by SabaunT — APPROVED** [review]

**Pull request merged**